sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/policy/StorageBearerTokenChallengeAuthorizationPolicy.java

```diff
@@ -65,15 +67,34 @@ public Mono<Boolean> authorizeRequestOnChallenge(HttpPipelineCallContext context
         Map<String, String> challenges = extractChallengeAttributes(authHeader, BEARER_TOKEN_PREFIX);
 
         String scope = challenges.get("resource_id");
-        if (scope != null) {
+        String authorization = challenges.get("authorization_uri");
+
+        if (scope != null && authorization != null) {
```

Comment:
authorization check should be in separate independent if block and then parse the tenant ID.
While its not common but its possible that authorization_uri is not present but scope/resource id is, need to accomodate this scenario.